app/src/ui/branches/ci-check-run-list.tsx

```diff
@@ -40,16 +43,47 @@ export class CICheckRunList extends React.PureComponent<
 
   public constructor(props: ICICheckRunListProps) {
     super(props)
+
+    const combinedCheck = props.dispatcher.tryGetCommitStatus(
+      this.props.repository,
+      this.getCommitRef(this.props.prNumber)
+    )
+
     this.state = {
-      check: props.dispatcher.tryGetCommitStatus(
+      checkRuns: combinedCheck !== null ? combinedCheck.checks : [],
+      checkRunsShown: null,
+      checkRunLogsShown: null,
+      loadingLogs: true,
+    }
+
+    this.onStatus(combinedCheck)
+  }
+
+  public componentDidUpdate(prevProps: ICICheckRunListProps) {
+    // Re-subscribe if we're being reused to show a different status.
+    if (
+      this.props.repository !== prevProps.repository ||
```

Comment:
Shouldn't we check the hash here? 🤔 
```suggestion
      this.props.repository.hash !== prevProps.repository.hash ||
```

Reply:
Hmm. Yes. I copied that from the badge output component.. Likely should be the hash there as well.

Reply:
I'm really not sure